src/client/tab.rs

```diff
@@ -268,11 +258,7 @@ impl Tab {
             if let Some(plugin) = &layout.plugin {
                 let (pid_tx, pid_rx) = channel();
                 self.send_plugin_instructions
-                    .send(PluginInstruction::Load(
-                        pid_tx,
-                        plugin.clone(),
-                        layout.events.clone(),
-                    ))
+                    .send(PluginInstruction::Load(pid_tx, plugin.clone()))
```

Comment:
Tbh, I don't like this approach.
It seems like we're trying to force a shared-state paradigm onto a distributed architecture. Instead of sending a channel over another channel and blocking until we receive a message back on it, can't we do something similar to what we're doing with terminals? We have the same problem there. There we load all of them beforehand. Do you think something like this can work with plugins too?

Reply:
I think the only tricky bit about this is knowing what "pid" to allocate. At the moment, only the plugin system is aware of how many plugins are loaded. I'm slightly hesitant to duplicate that state, but I'm open to thinking that through some more?

I've been using this system since the initial implementation of the plugin system without too many issues, but I certainly agree that it's not ideal. Are you suggesting that I allocate a PID in `tab.rs` then send that in the loading message?

Reply:
Not exactly. I'm suggesting we do the same thing we do with terminals. We spawn all of them before we get to this function, and then pass their pids. What happens (roughly) is: when we want to open a new tab, we send a `NewTab` instruction to the pty thread. The pty thread counts all the terminals that need to be spawned, spawns them and then sends the `ApplyLayout` instruction to screen already including the terminal pids. I'm suggesting we add another link to this chain (either before or after the pty thread) where we go through the plugin thread, spawn the necessary plugins, tack on their pids to the instruction and forward everything to screen.
So that in the end, this function would get `new_terminal_pids` and `new_plugin_pids` parameters. Makes sense?

Reply:
Ah, okay, I see what you mean now! I'll take a look at how I might be able to make that happen!